apps/wallet/src/ui/app/staking/stake/StakeForm.tsx

```diff
@@ -46,48 +49,111 @@ function StakeForm({
 
     const [formatted, symbol] = useFormatCoin(coinBalance, coinType);
 
+    const allCoins = useAppSelector(accountCoinsSelector);
+    const allCoinsOfTransferType = useMemo(
+        () => allCoins.filter((aCoin) => aCoin.type === coinType),
+        [allCoins, coinType]
+    );
+    const [coinDecimals] = useCoinDecimals(coinType);
+    const gasEstimationUnformatted = useMemo(() => {
+        return Coin.computeGasBudgetForPay(
+            allCoinsOfTransferType,
+            parseAmount(amount, coinDecimals)
+        );
+    }, [allCoinsOfTransferType, amount, coinDecimals]);
+
+    const [gasBudgetEstimation] = useFormatCoin(
+        gasEstimationUnformatted,
+        coinType
+    );
+
+    const maxToken = formatBalance(coinBalance, coinDecimals);
+    const setMaxToken = useCallback(() => {
+        setFieldValue('amount', maxToken);
+    }, [maxToken, setFieldValue]);
+
     return (
-        <Form className={st.container} autoComplete="off" noValidate={true}>
-            <div className={st.group}>
-                <label className={st.label}>Amount:</label>
-                <Field
-                    component={NumberInput}
-                    allowNegative={false}
-                    name="amount"
-                    placeholder={`Total ${symbol} to stake`}
-                    className={st.input}
-                    decimals
-                />
-                <div className={st.muted}>
-                    Available balance: {formatted} {symbol}
-                </div>
+        <Form
+            className="flex flex-1 flex-col flex-nowrap"
+            autoComplete="off"
+            noValidate={true}
+        >
+            <Content>
+                <Card
+                    variant="blue"
+                    header={
+                        <div className="p-2.5 w-full flex bg-white">
+                            <Field
+                                component={NumberInput}
+                                allowNegative={false}
+                                name="amount"
+                                className="w-full border-none text-hero-dark text-heading4 font-semibold placeholder:text-gray-70 placeholder:font-medium"
+                                decimals
+                            />
+                            <div
+                                role="button"
+                                className="border border-solid border-gray-60 hover:border-steel-dark rounded-2xl h-6 w-11 flex justify-center items-center cursor-pointer text-steel-darker hover:text-steel-darker text-bodySmall font-medium"
+                                onClick={setMaxToken}
```

Comment:
I think we should disable setMatToken button until `maxToken` above is ready. We fetch coin denomination data